Assets/Scripts/Buildings/Distribution/Market.cs

```diff
@@ -3,11 +3,38 @@
 using UnityEngine;
 
 public class Market : DistributionBuilding
-{
-	private Resources tempResources;
-	
-	public void AddResourcess(Resources newResources)
-	{
-		BuildManager.instance.playerResources += newResources;
-	}
+{	
+	void Update()
+   {
+		if(timeFromLastCheck < 0f) 
+		{
+			if(lastResources.fish >= resourcesOnMarket.fish)
```

Comment:
lastResources i resourcesOnMarket have to be initialized. Otherwise you NullReferenceException